cmd/lastseen.go

```diff
@@ -16,38 +17,56 @@ func LastSeen(msg *Message) {
 	if len(msg.Params) < 2 {
 		msg.Return("Usage: .last [nick]")
 	} else {
-		nick := msg.Params[1]
-		seen := GetLastSeen(nick)
-
-		msg.Return(fmt.Sprintf("%s was last seen %s", nick, seen))
+		seen := GetLastSeen(msg)
+		msg.Return(fmt.Sprintf("%s was last seen %s", msg.Params[1], seen))
 	}
 }
 
-func GetLastSeen(nick string) string {
+func GetLastSeen(msg *Message) string {
+	nick := msg.Params[1]
+	channel := msg.To
+	key := fmt.Sprintf("%s%s", nick, channel)
+	msgkey := fmt.Sprintf("%s:msg", key)
+
 	conn := pool.Get()
 	defer conn.Close()
-
 	cmd := fmt.Sprintf("%s:%s", PREFIX, "lastseen")
-	seen, err := redis.Int64(conn.Do("hget", cmd, nick))
 
+	seen, err := redis.Int64(conn.Do("hget", cmd, key))
 	if err != nil {
 		return "never"
 	}
+	seenstr := humanize.Time(time.Unix(seen, 0))
+
+	lastmsg, err2 := redis.String(conn.Do("hget", cmd, msgkey))
```

Comment:
Because we're guaranteed that `err` is nil here (or we would have returned "never"), we don't need to use `err2`, but can just use `lastmsg, err :=`